torch/_subclasses/fake_impls.py

```diff
@@ -1029,7 +1029,7 @@ def slow(msg):
 def fast_detach(fake_mode, x):
     with no_python_dispatcher(), in_kernel_invocation_manager(fake_mode):
         out = torch.ops.aten.detach.default(x)
-    return FakeTensor(fake_mode, out, x.device)
+    return FakeTensor(fake_mode, out, x.device, real_tensor=x.real_tensor)
```

Comment:
Btw, this is wrong - the real tensor should be out.real_tensor

Reply:
> Btw, this looks wrong - the real tensor should be out.real_tensor
@zou3519 Do you mean the real tensor should be `out`? I think `out` here is already a real tensor.

Reply:
My reading of the code is that `x` is the input to detach and we are trying to create a FakeTensor that is the output of the detach. Why are we using the input's real tensor value as the new FakeTensor's real tensor value?

Reply:
> My reading of the code is that `x` is the input to detach and we are trying to create a FakeTensor that is the output of the detach. Why are we using the input's real tensor value as the new FakeTensor's real tensor value?

@zou3519  The way it's used is like this: we want a snapshot of the FakeTensor `x`. And it seems like the way we do it is using `snapshot_fake` to create a snapshot of `x` (this is where the detach is called). IMO, the new snapshot FakeTensor should have the same underlying real tensor as `x`.